### DISCUSSION TITLE
Are you planning to use a Markdown-based editor?

### DISCUSSION THREAD

**Discussion by ndt080** [pocketbase/pocketbase#3429] [Q&A] [answered]
Are you planning to use a Markdown-based editor? Using HTML-based editors is unsafe, as it creates vulnerabilities for XSS (Cross-Site Scripting) attacks. The second problem I found is loading images as a Blob.  Pocketbase does not have a single mechanism for managing all previously uploaded media data, which creates the problem of duplication of media data and the inability to get an exact link to the media file that could be inserted into the editor. Therefore, the only option is to use a Blob, which increases the size of the returned JSON to an unthinkable scale

**Comment by ganigeorgiev** [comment, marked as answer]
For now there are no plans to change the editor (TinyMCE has markdown support in their [roadmap](https://www.tiny.cloud/roadmap/)).

About the images upload - with the next release there are plans to add collections image picker as part of [task#37830705](https://github.com/orgs/pocketbase/projects/2/views/1?pane=issue&itemId=37830705).

**Reply by ndt080** [reply]
That sounds encouraging. Is it possible to install your own plugins for TinyMCE in the "Use as a framework" mode?

**Reply by ganigeorgiev** [reply]
No, the Admin UI is not extendable at the moment.

**Reply by ndt080** [reply]
![image](https://github.com/pocketbase/pocketbase/assets/68751397/f288d6b1-ba99-4a16-8b11-96560f84ced6)

**Reply by ganigeorgiev** [reply]
_Just for info in case someone stumble here - support for "Media library"-like functionality was added in the `develop` branch. It works by listing all non-protected record files from a specific collection (it support collections with more than one single and/or multiple `file` fields). The UI currently looks something like this (labels are subject to change):_

![editor_dropdown](https://i.imgur.com/eIRgKNE.png)

![picker](https://i.imgur.com/bTclI9O.png)
